<!DOCTYPE html>

<html lang="en" xmlns="http://www.w3.org/1999/xhtml">
<head>
    <meta charset="utf-8" />
    <title>E-Portfolio Bioinformatics Workflows Research</title>
    <!--Connecting html and css files-->
    <link rel="stylesheet" href="style.css">
</head>
<body>
    <div id="header">
        <div class="container">
            <!--Navigation back to main research page-->
            <nav class="res-link-nav">
                <ul>
                    <li><a href="research.html">Back to Main Research Page</a></li>
                </ul>
            </nav>
            <div id="research-page">
                <div class="container">
                    <h1 class="subtitle">
                        Creating Easy-To-Use Bioinformatics Templates
                    </h1>
                    <h2 class="sub-subtitle">
                        <br />
                        May 2025 - Present
                        <br />
                        <br />
                        Principle Investigator: <span class="bold-text">Dr. Arun Seetharam, Dr. Nannan Shan</span> 
                        <br />
                        Institution: <span class="bold-text">Purdue University</span>
                        <br />
                        Department: <span class="bold-text">Rosen Center For Advanced Computing</span>
                        <br />
                    </h2>
                    <h2 class="sub-subtitle">
                        <br />
                        Research Focus:
                    </h2>
                    <p class="res-paragraphs">
                            This research project is focused on lowering the barrier for biologists to use HPC for data analysis.
                            Currently, tools like NextFlow and SnakeMake do exist to create workflows, however they are complex
                            and difficult to learn. For instance, both tools require the user to learn a Domain Specific Language (DSL).
                            <br /><br />
                            Meanwhile our workflow only requires the user to know Python and Bash, which are both popular languages
                            that is it very likely the user already knows. Specifically, we have created workflows for Hifiasm and Flye
                            Genome Assembly, GATK Variant Calling, and RNA-seq.
                        </p>
                    <h2 class="sub-subtitle">
                        <br />
                        Responsibilities:
                    </h2>
                    <p class="res-paragraphs">
                        I am working with both Dr. Arun Seetharam and Dr. Nannan Shan at Purdue University's Rosen Center For Advanced
                        Computing (RCAC) in a summer REU internship program. The project's overall PI is Arman Pazouki, who holds the 
                        NSF grant.
                        <br /><br />
                        In this project, my responsibilities centered around programming the Hifiasm and Variant Calling workflows using
                        a combination of Bash scripts, Python, and the tool Urwid (to create our Console User Interface (CUI)).
                        <br /><br />
                        I am also creating a general template to allow users to automatically generate a CUI from a pipeline.
                        <br /><br />
                    </p>
                    <br />
                    <p class="caption">
                        Workflows
                    </p>
                    <div class="ot2-img">
                        <br />
                        <img src="images/ot-2.png" />
                    </div>
                    <p class="caption">
                        Visual of OT-2 Liquid Pipetting Precision
                    </p>
                    <div class="ot2-img">
                        <br />
                        <img src="images/ot-2_precision.jpg" />
                    </div>






                </div>
            </div>
        </div>
    </div>


</body>
</html>

<!--Common molecular biology techniques such as DNA assembly, plasmid extraction and bacterial transformation involve labor-intensive
                        manual processes and are currently difficult to scale. These processes become tedious and more error-prone when handling many samples
                        at once. To address these notable challenges and accelerate discoveries in plant synthetic biology and related disciplines, we are deploying
                        Opentrons OT-2 liquid-handling robots to automate these procedures. The Designer Glycans lab is developing Python protocols to adapt
                        various experimental steps for the OT-2 system. Specifically, I designed a small volume transfer protocol for DNA manipulations and
                        amplifications allowing biologists to customize sample volumes, positions, and pipetting parameters in a user-friendly table, catering to
                        diverse scientific needs. Moreover, the OT-2 robots can also facilitate more intricate processes like plasmid extraction using 96-well
                        plates and modular equipment such as programmable magnetic blocks. Overall, our implementation of liquid handling robots into otherwise
                        artisanal procedures will reduce scientific barriers and minimize errors for biological research and product development.-->
